src/FAI3_backend/src/types.rs

```diff
@@ -81,9 +81,95 @@ pub struct ModelDetails {
 pub struct User {
     pub(crate) user_id: Principal,
     pub(crate) models: Vec<u128>,
+    pub(crate) llm_models: Vec<u128>,
 }
 
 #[derive(Serialize, Deserialize, Debug)]
 pub struct HuggingFaceResponseItem {
     pub(crate) generated_text: Option<String>,
 }
+
+// LLMs
+#[derive(Serialize, Copy, Clone, PartialEq, Debug, CandidType, CandidDeserialize)]
+pub enum ContextAssociationTestResult {
+    Stereotype,
+    AntiStereotype,
+    Neutral,
+    Other,
+}
+
+#[derive(Serialize, Deserialize, CandidType, Clone, Debug)]
+pub struct ContextAssociationTestMetrics {
+    pub(crate) stereotype: i32,
+    pub(crate) anti_stereotype: i32,
+    pub(crate) neutral: i32,
+    pub(crate) other: i32,
+}
+
+#[derive(Serialize, Deserialize, CandidType, Clone, Debug)]
+pub struct ContextAssociationTestMetricsBag {
+    pub(crate) general: ContextAssociationTestMetrics,
+    pub(crate) intersentence: ContextAssociationTestMetrics,
+    pub(crate) intrasentence: ContextAssociationTestMetrics,
+    pub(crate) gender: ContextAssociationTestMetrics,
+    pub(crate) race: ContextAssociationTestMetrics,
+    pub(crate) religion: ContextAssociationTestMetrics,
+    pub(crate) profession: ContextAssociationTestMetrics,
+    pub(crate) error_count: i32,
+    pub(crate) timestamp: u64,
+    pub(crate) intrasentence_prompt_template: String,
+    pub(crate) intersentence_prompt_template: String,
+    pub(crate) seed: i32,
+    // precalculated fields
+    pub(crate) icat_score_intra: f32,
+    pub(crate) icat_score_inter: f32,
+    pub(crate) icat_score_gender: f32,
+    pub(crate) icat_score_race: f32,
+    pub(crate) icat_score_profession: f32,
+    pub(crate) icat_score_religion: f32,
+    pub(crate) general_lms: f32,
+    pub(crate) general_ss: f32,
+    pub(crate) general_n: i32,
+    pub(crate) icat_score_general: f32,
+    pub(crate) data_points: Vec<ContextAssociationTestDataPoint>,
```

Comment:
We might want this to be optional so it can be set to null when requesting it through the API to avoid unnecessary traffic.